Ghidra/Features/PDB/src/pdb/cpp/iterate.cpp

```diff
@@ -541,64 +532,66 @@ void iterateFrameData(IDiaEnumFrameData * pEnumFrameData) {
 	}
 }
 
-int iterateTables(const bool printAll) {
-	printf("%s<tables>\n", indent(4));
-	HRESULT hr;
-	DWORD celt;
+int iterateTables(PDBApiContext& ctx, bool printAll) {
+	printf("%S<tables>\n", indent(4).c_str());
+	HRESULT hr = S_OK;
+	DWORD celt = 0;
 
-	IDiaEnumTables * pTables;
+    CComPtr<IDiaEnumTables> pTables;
 
-	hr = pSession->getEnumTables( &pTables );
-	if ( hr < 0) {
+	hr = ctx.Session().getEnumTables( &pTables );
```

Comment:
lines 542-544: eliminate hr (and on line 537), and do if (FAILED(ctx.Session()...) {
        return; (without hr return)